File: s-recursos/src/main/java/co/edu/javeriana/arqui/rest/controller/MedicamentoController.java
package co.edu.javeriana.arqui.rest.controller;

import java.util.List;

import co.edu.javeriana.arqui.rest.model.Medicamento;
import co.edu.javeriana.arqui.rest.service.MedicamentoService;
import jakarta.inject.Inject;
import jakarta.ws.rs.Consumes;
import jakarta.ws.rs.GET;
import jakarta.ws.rs.POST;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;

@Path("/medicamentos")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
public class MedicamentoController {

    @Inject
    MedicamentoService service;

    @POST
    public Medicamento registrar(Medicamento m) {
        return service.registrar(m);
    }

    @GET
    public List<Medicamento> listar() {
        return service.listar();
    }
}

File: ms-medicamentos/src/main/kotlin/co/edu/javeriana/arqui/controller/MedicamentoController.kt
package co.edu.javeriana.arqui.controller

import co.edu.javeriana.arqui.model.Medicamento
import co.edu.javeriana.arqui.repository.MedicamentoRepository
import jakarta.inject.Inject
import jakarta.ws.rs.*
import jakarta.ws.rs.core.MediaType
import jakarta.ws.rs.core.Response

@Path("/medicamentos")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
class MedicamentoController {

    @Inject
    lateinit var repository: MedicamentoRepository

    @POST
    fun addMedicamento(medicamento: Medicamento): Response {
        repository.save(medicamento)
        return Response.status(Response.Status.CREATED).entity(medicamento).build()
    }

    @GET
    fun getAllMedicamentos(): List<Medicamento> = repository.findAll()
}
